cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -370,16 +372,22 @@ protected void maybeUpdateTopicConfig() {
 
   protected void maybeIncreaseTopicPartitionCount() {
     String cruiseControlMetricsTopic = _metricsTopic.name();
+
     try {
-      // Retrieve topic partition count to check and update.
-      TopicDescription topicDescription =
-          _adminClient.describeTopics(Collections.singletonList(cruiseControlMetricsTopic)).values()
-                      .get(cruiseControlMetricsTopic).get(CLIENT_REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+      Method topicDescriptionMethod = topicNameValuesMethod();
```

Comment:
It may be worth adding a comment right above this line and other places where we call `topicNameValuesMethod()`, e.g. "// For compatibility with Kafka 4.0 and beyond we must use new API methods."